### PR TITLE
Add "container-runtime-endpoint" and "image-service-endpoint" to kubelet config

### PR DESCRIPTION
**Pull request by adisky**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add "container-runtime-endpoint" and "image-service-endpoint" to kubelet config, These kubelet flags were marked as experimental before so not added in kubelet config, with dockershim going away these flags would be used by more users and should be available via config file as well

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
Signed-off-by: Aditi Sharma <adi.sky17@gmail.com>
/sig node

### CODE REVIEW

File: pkg/kubelet/apis/config/types.go [hunk 1]
```diff
@@ -444,6 +444,16 @@ type KubeletConfiguration struct {
 	// registerNode enables automatic registration with the apiserver.
 	// +optional
 	RegisterNode bool
+
+	// containerRuntimeEndpoint is the endpoint of container runtime.
+	// unix domain sockets supported on Linux while npipes and tcp endpoints are supported for windows.
+	// examples:'unix:///path/to/runtime.sock', 'npipe:////./pipe/runtime
```
Comment (ehashman) [inline pkg/kubelet/apis/config/types.go:450]:
```suggestion
	// Examples: 'unix:///path/to/runtime.sock', 'npipe:////./pipe/runtime'
```

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -282,24 +282,21 @@ func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, ku
...
-	kubeDeps *Dependencies,
-	remoteRuntimeEndpoint string,
-	remoteImageEndpoint string) error {
-	// remoteImageEndpoint is same as remoteRuntimeEndpoint if not explicitly specified
-	if remoteRuntimeEndpoint != "" && remoteImageEndpoint == "" {
-		remoteImageEndpoint = remoteRuntimeEndpoint
+func PreInitRuntimeService(kubeCfg *kubeletconfiginternal.KubeletConfiguration, kubeDeps *Dependencies) error {
+	// imageServiceEndpoint is same as containerRuntimeEndpoint if not explicitly specified
+	if kubeCfg.ContainerRuntimeEndpoint != "" && kubeCfg.ImageServiceEndpoint == "" {
+		kubeCfg.ImageServiceEndpoint = kubeCfg.ContainerRuntimeEndpoint
```
Comment (ehashman) [inline pkg/kubelet/kubelet.go:288]: Will this mutate the kubeCfg? Why not just declare a local var?

File: staging/src/k8s.io/kubelet/config/v1beta1/types.go [hunk 3]
```diff
@@ -778,6 +778,16 @@ type KubeletConfiguration struct {
 	// Default: true
 	// +optional
 	RegisterNode *bool `json:"registerNode,omitempty"`
+
+	// containerRuntimeEndpoint is the endpoint of container runtime.
+	// unix domain sockets supported on Linux while npipes and tcp endpoints are supported for windows.
+	// examples:'unix:///path/to/runtime.sock', 'npipe:////./pipe/runtime
```
Comment (ehashman) [inline staging/src/k8s.io/kubelet/config/v1beta1/types.go:784]:
```suggestion
	// Examples: 'unix:///path/to/runtime.sock', 'npipe:////./pipe/runtime'
```

File: staging/src/k8s.io/kubelet/config/v1beta1/types.go [hunk 4]
```diff
@@ -778,6 +778,16 @@ type KubeletConfiguration struct {
...
+	ContainerRuntimeEndpoint string `json:"containerRuntimeEndpoint,omitempty"`
```
Comment (ehashman) [inline staging/src/k8s.io/kubelet/config/v1beta1/types.go:785]: Only optional fields should have an `omitempty` (see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#edit-typesgo)

What happens if we leave this empty? This is a required field, right? We don't set up any defaulting in this PR from what I can see.

**Review by ehashman — COMMENTED** [review]
First pass API review, mostly looks good, a few fixes needed...

**Comment by ehashman** [comment]
/priority important-soon

Important for dockershim deprecation

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (adisky) [reply pkg/kubelet/kubelet.go]: done

**Comment by pacoxu** [comment]
@adisky  do you have time to update this pr? https://github.com/kubernetes/kubeadm/issues/1924 is pending on this one.

**Pull request closed without merging**